compendium/tools/SCRCodeGen/ManifestParserImpl.cpp

```diff
@@ -103,6 +104,13 @@ ManifestParserImplV1::ParseAndGetComponentInfos(Json::Value const& scr) const
         {
             auto factoryComponentID
                 = JsonValueValidator(jsonComponent, "factory", Json::ValueType::stringValue).GetString();
+            if (!factoryComponentID.empty()) {
+                if (duplicatePids.size() != 1)
```

Comment:
reduce nested if statements by combining the two if statements into one.

Reply:
Will fix